### PR TITLE
Fix display value for selected filters in study view

### PR DESCRIPTION
**Pull request by aswin2108**
Fix cBioPortal/cbioportal#9266

## Screenshots
**Before:** 
image

**After:**
image

image

### CODE REVIEW

**Review by onursumer — CHANGES_REQUESTED** [review]
This breaks some already existing unit test cases. You should be able to fix the issue without introducing new bugs. And ideally we should also add new unit tests for the case you are trying to fix.

```
Summary of all failing tests
 FAIL  src/pages/studyView/StudyViewUtils.spec.tsx (45.768 s)
  ● StudyViewUtils › intervalFiltersDisplayValue › generates display value for filter values with start closed, end open

    assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      "> 10"
    Received:
      "≥ 10"
    
    Message:
      expected '≥ 10' to equal '> 10'

    Difference:

    - Expected
    + Received

    - > 10
    + ≥ 10

      1424 |                 true
      1425 |             );
    > 1426 |             assert.equal(value, '> 10');
           |                    ^
      1427 |         });
      1428 |
      1429 |         it('generates display value for filter values with start closed, end open, with special values', () => {

      at Object.<anonymous> (src/pages/studyView/StudyViewUtils.spec.tsx:1426:20)

  ● StudyViewUtils › intervalFiltersDisplayValue › generates display value for filter values with start closed, end open, with special values

    assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      "> 10, NA, REDACTED"
    Received:
      "≥ 10, NA, REDACTED"
    
    Message:
      expected '≥ 10, NA, REDACTED' to equal '> 10, NA, REDACTED'

    Difference:

    - Expected
    + Received

    - > 10, NA, REDACTED
    + ≥ 10, NA, REDACTED

      1433 |                 true
      1434 |             );
    > 1435 |             assert.equal(value, '> 10, NA, REDACTED');
           |                    ^
      1436 |         });
      1437 |
      1438 |         it('generates display value for filter values with special values only', () => {

      at Object.<anonymous> (src/pages/studyView/StudyViewUtils.spec.tsx:1435:20)


Test Suites: 1 failed, 3 skipped, 184 passed, 185 of 188 total
Tests:       2 failed, 13 skipped, 1786 passed, 1801 total
```

**Review by onursumer — CHANGES_REQUESTED** [review]

**Review by onursumer — CHANGES_REQUESTED** [review]

File: src/pages/studyView/StudyViewUtils.tsx [hunk 1]
```diff
@@ -1586,6 +1586,12 @@ export function intervalFiltersDisplayValue(
         } else if (start === undefined) {
             intervalDisplayValues.push(`≤ `);
             intervalDisplayValues.push(endText);
+        } else if (
+            end === undefined &&
+            numericals[0].start === numericals[0].end
+        ) {
+            intervalDisplayValues.push(`≥ `);
+            intervalDisplayValues.push(startText);
```
Comment (onursumer) [inline src/pages/studyView/StudyViewUtils.tsx:1594]: Looks better. Can we move this condition directly into `else if (end === undefined)` block?

Something like this:
```suggestion
        } else if (
            end === undefined
        ) {
            if (numericals[0].start === numericals[0].end) {
                intervalDisplayValues.push(`≥ `);
            } else {
                intervalDisplayValues.push(`> `);
            }
            
            intervalDisplayValues.push(startText);
```

**Review by onursumer — CHANGES_REQUESTED** [review]
Looks better. Can you also add a new test case to cover the bug you are fixing?

**Review by onursumer — COMMENTED** [review]
Looks good! Can you rebase to the latest master and squash commits to a single commit? (`git rebase -i upstream/master`)

File: src/pages/studyView/StudyViewUtils.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aswin2108) [reply src/pages/studyView/StudyViewUtils.tsx]: Actually, I noticed your suggestion after pushing the test, so just after that itself, I pushed these changes!

[the remark this answers is not in the document]
Comment (aswin2108) [reply src/pages/studyView/StudyViewUtils.tsx]: @onursumer I have squashed and rebased, what to do next?

**Review by onursumer — APPROVED** [review]
Looks good. We may need some manual product testing before merging this PR to make sure that nothing has been accidentally broken.

**Review by dippindots — APPROVED** [review]
Looks good to me from code side. @aswin2108 Thanks for fixing this.

**Review by inodb — APPROVED** [review]
Great work 🏆! Thanks so much for your contribution!

**Pull request merged**